docs/source/java/modules.rst

```diff
@@ -0,0 +1,413 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+.. or more contributor license agreements.  See the NOTICE file
+.. distributed with this work for additional information
+.. regarding copyright ownership.  The ASF licenses this file
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+.. default-domain:: java
+.. highlight:: java
+
+.. _java_modules:
+
+=======
+Modules
+=======
+
+.. contents::
+
+This is the variety of arrow java documentation:
+
+* `Specification and protocols`_: This contains agnostic specification that is implemented in this case by arrow java modules.
+* `Supported environment`_ (like this): This contains answers for what-is-that arrow java module.
+* `Cookbook`_: This contains answers about how-to-use arrow java modules with practices examples.
```

Comment:
IMO, we don't need most of these links (they're part of the sidebar already). But we can link to the cookbook in the actual getting started page.

Reply:
Changed